### ISSUE TITLE
`keymap` fails to get parent keymap from parent scheme

### ISSUE THREAD

**Issue by Ambrevar** [atlas-engineer/nyxt#1405]
In
https://github.com/atlas-engineer/nyxt/pull/1404
I added a binding, `C-R` to the CUA scheme.

Notice that if you use the Emacs or VI scheme, it won't work :(

Adding an empty Emacs or VI keymap fixes it:

```lisp
(define-mode auto-mode ()
  "...."
  (...
   (keymap-scheme
    (define-scheme "auto-mode"
      scheme:cua
      (list
       "C-R" 'reload-with-modes)
      scheme:emacs
      (list)
      scheme:vi
      (list)))
 ...)
```

This is obviously an ugly workaround.
The issue is with the `keymap` method and indirectly `keymap:get-keymap`:

```lisp
(keymap:get-keymap (if (buffer mode)
                         (keymap-scheme-name (buffer mode))
                         scheme:cua)
                     (keymap-scheme mode))

; keymap/scheme.lisp:
(defun get-keymap (name scheme)
  "Return keymap corresponding to NAME in SCHEME.
Return nil if not found."
  (gethash name scheme))
```

Looks like this is wrong, it should return the keymap of the parent scheme if SCHEME has no keymap.

I'll need to think twice about it.

Crazy we've never hit this bug before!

**Comment by Ambrevar** [comment]
Fixed with 8011f7847e4bd28fd48a05abd7e9ca7904271b1e.

**Comment by jmercouris** [comment]
Wooooooo!!!!!!!!
